aws/resource_aws_transfer_user_test.go

```diff
@@ -125,6 +125,37 @@ func TestAccAWSTransferUser_disappears(t *testing.T) {
 	})
 }
 
+func TestAccAWSTransferUserName_validation(t *testing.T) {
```

Comment:
Nit: The acceptance test naming format should match: `TestAccAWSTransferUser_XXX` so all testing for the resource can be run with the same prefix and underscore, e.g. `TestAccAWSTransferUser_UserName_Validation`

Otherwise we have this situation (note lack of this new test being run):

```console
$ TF_ACC=1 go test ./aws -v -timeout 120m -parallel 20 -run='TestAccAWSTransferUser_'
=== RUN   TestAccAWSTransferUser_basic
=== PAUSE TestAccAWSTransferUser_basic
=== RUN   TestAccAWSTransferUser_modifyWithOptions
=== PAUSE TestAccAWSTransferUser_modifyWithOptions
=== RUN   TestAccAWSTransferUser_disappears
=== PAUSE TestAccAWSTransferUser_disappears
=== CONT  TestAccAWSTransferUser_basic
=== CONT  TestAccAWSTransferUser_modifyWithOptions
=== CONT  TestAccAWSTransferUser_disappears
--- PASS: TestAccAWSTransferUser_disappears (11.24s)
--- PASS: TestAccAWSTransferUser_basic (12.36s)
--- PASS: TestAccAWSTransferUser_modifyWithOptions (26.95s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	27.993s
```

Reply:
Okay, I'll update and make another PR. Thanks for the review!